File: 6/sender/main.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"log"
	"net/smtp"
	"time"
)

// Config описывает структуру конфигурационного файла.
type Config struct {
	SMTPHost      string `json:"smtpHost"`
	SMTPPort      string `json:"smtpPort"`
	IMAPServer    string `json:"imapServer"`
	Username      string `json:"username"`
	Password      string `json:"password"`
	From          string `json:"from"`
	To            string `json:"to"`
	LabName       string `json:"labName"`
	SenderName    string `json:"senderName"`
	SenderGroup   string `json:"senderGroup"`
	ReceiverName  string `json:"receiverName"`
	ReceiverGroup string `json:"receiverGroup"`
}

func loadConfig(filename string) (*Config, error) {
	data, err := ioutil.ReadFile(filename)
	if err != nil {
		return nil, err
	}
	var config Config
	if err := json.Unmarshal(data, &config); err != nil {
		return nil, err
	}
	return &config, nil
}

func sendEmail(config *Config) error {
	auth := smtp.PlainAuth("", config.Username, config.Password, config.SMTPHost)
	now := time.Now().Format(time.RFC1123Z)
	subject := config.LabName

	msg := "From: " + config.From + "\r\n" +
		"To: " + config.To + "\r\n" +
		"Subject: " + subject + "\r\n" +
		"X-Priority: 1\r\n" +
		"Importance: High\r\n" +
		"\r\n" +
		"Отправитель: " + config.SenderName + ", группа " + config.SenderGroup + "\r\n" +
		"Получатель: " + config.ReceiverName + ", группа " + config.ReceiverGroup + "\r\n" +
		"Время отправки: " + now + "\r\n"

	if err := smtp.SendMail(config.SMTPHost+":"+config.SMTPPort, auth, config.From, []string{config.To}, []byte(msg)); err != nil {
		return err
	}
	fmt.Println("Письмо отправлено успешно!")
	return nil
}

func main() {
	config, err := loadConfig("../config.json")
	if err != nil {
		log.Fatal("Ошибка загрузки конфигурации:", err)
	}

	if err := sendEmail(config); err != nil {
		log.Fatal("Ошибка при отправке письма:", err)
	}
}

File: 6/python/sender/main.py
import json
import smtplib
import ssl
import datetime
from email.message import EmailMessage

def load_config(filename):
    with open(filename, "r", encoding="utf-8") as f:
        return json.load(f)

def send_email(config):
    msg = EmailMessage()
    msg["From"] = config["from"]
    msg["To"] = config["to"]
    msg["Subject"] = config["labName"]
    msg["X-Priority"] = "1"
    msg["Importance"] = "High"

    now = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
    body = (
        f"Отправитель: {config['senderName']}, группа {config['senderGroup']}\n"
        f"Получатель: {config['receiverName']}, группа {config['receiverGroup']}\n"
        f"Время отправки: {now}\n"
    )
    msg.set_content(body)

    smtp_host = config["smtpHost"]
    smtp_port = int(config["smtpPort"])
    username = config["username"]
    password = config["password"]

    context = ssl.create_default_context()
    with smtplib.SMTP(smtp_host, smtp_port) as server:
        server.starttls(context=context)
        server.login(username, password)
        server.send_message(msg)
    print("Письмо отправлено успешно!")

if __name__ == '__main__':
    config = load_config("../../config.json")
    send_email(config)
